blazor/rich-text-editor/webassembly-performance.md

```diff
@@ -0,0 +1,139 @@
+
+* **OnToolbarClick** event is bound to a callback method, when the editor content gets changed by toolbar action the event is completed, the **StateHasChanged** will be invoked for the parent component.
+
+```cshtml
+@using Syncfusion.Blazor.RichTextEditor
+
+<SfRichTextEditor @ref="rteObj">
+    <RichTextEditorToolbarSettings Items="@Tools" Type="ToolbarType.Expand" />
+    <RichTextEditorEvents OnToolbarClick="@ToolbarClick"/>
+</SfRichTextEditor>
+<div>
+    <span>@((MarkupString)Output)</span>
+</div>
+    
+
+@code {
+    SfRichTextEditor rteObj;
+    private string Output = "";
+    private void ToolbarClick(ToolbarClickEventArgs args)
+    {
+        rteObj.PreventRender();
+        this.Output = this.Output + "<span><b>OnToolbarClick</b> event called<hr></span>";
+    }
+    private List<ToolbarItemModel> Tools = new List<ToolbarItemModel>() 
+    { 
+        new ToolbarItemModel() { Command = ToolbarCommand.Bold }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.Italic }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.Underline }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.StrikeThrough }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.Separator }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.FontColor }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.BackgroundColor }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.Separator }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.Formats }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.Alignments }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.Separator }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.OrderedList }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.UnorderedList }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.Separator }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.Outdent }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.Indent }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.Separator },
+        new ToolbarItemModel() { Command = ToolbarCommand.CreateLink }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.Image }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.CreateTable }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.Separator }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.Undo }, 
+        new ToolbarItemModel() { Command = ToolbarCommand.Redo } 
+    }; 
+}
+```
+
+N> * **PreventRender** method internally overrides the **ShouldRender** method of the RichTextEditor to prevent rendering.
+<br/> * It is recommended to use the **PreventRender** method for user interactive events such as OnToolbarClick, UpdatedToolbarStatus, etc., for better performance.
+<br/> * For events without any argument such as **DataBound**, you can use the **PreventRender** method of the RichTextEditor to disable rendering.
```

Comment:
![image](https://github.com/syncfusion-content/blazor-docs/assets/29145454/8097ad5c-d0be-45cf-b3b1-eb410e10e9a1)


DataBound, we dont have any arguments